src/operations/Operation.cpp

```diff
@@ -249,4 +249,29 @@ void Operation::addDepthContribution(std::vector<std::size_t>& depths) const {
   }
 }
 
+void Operation::addTwoQubitDepthContribution(
```

Comment:
This doesn't just count two-qubit gates but any gate with a control. So a multi-controlled x-gate would still be considered a two-qubit gate in this case. I actually don't think this is a huge problem but the feature and the methods should be renamed accordingly (i.e. multiQubitGateDepth or something).